citesphere/src/main/java/edu/asu/diging/citesphere/core/authority/impl/ConceptpowerImporter.java

```diff
@@ -0,0 +1,158 @@
+     * retrieveAuthorityData(java.lang.String)
+     */
+    @Override
+    public IImportedAuthority retrieveAuthorityData(String searchString)
+            throws UnsupportedOperationException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public AuthoritySearchResult searchAuthorities(String firstName, String lastName, int page, int pageSize)
+            throws AuthorityServiceConnectionException {
+
+        RequestEntity<Void> request;
+        try {
+
+            String url = conceptpowerURL + conceptpowerSearchKeyword + URLEncoder
+                    .encode(this.getConceptpowerSearchString(firstName, lastName), StandardCharsets.UTF_8.toString())
+                    + "&page=" + page;
+            URI uri = UriComponentsBuilder.fromUriString(url.toString()).build(true).toUri();
+
+            request = RequestEntity.get(uri).accept(MediaType.APPLICATION_JSON).build();
+        } catch (UnsupportedEncodingException e) {
+            throw new AuthorityServiceConnectionException("Exception occured while creating URI for authority search",
+                    e);
+        }
+        ResponseEntity<ConceptpowerResponse> response = null;
+        try {
+            response = restTemplate.exchange(request, ConceptpowerResponse.class);
+        } catch (RestClientException ex) {
+            throw new AuthorityServiceConnectionException(ex);
+        }
+
+        List<IAuthorityEntry> authorityEntries = new ArrayList<IAuthorityEntry>();
+        AuthoritySearchResult searchResult = new AuthoritySearchResult();
+        if (response.getStatusCode() == HttpStatus.OK) {
+            ConceptpowerResponse conceptEntries = response.getBody();
+            if (conceptEntries.getConceptEntries() != null) {
+                for (ConceptpowerEntry conceptEntry : conceptEntries.getConceptEntries()) {
+                    IAuthorityEntry authority = new AuthorityEntry();
+                    authority.setName(conceptEntry.getLemma());
+                    authority.setUri(conceptEntry.getConcept_uri());
+                    authority.setDescription(conceptEntry.getDescription());
+                    authorityEntries.add(authority);
+                }
+                searchResult.setFoundAuthorities(authorityEntries);
+                searchResult.setTotalPages((int) Math
+                        .ceil(conceptEntries.getPagination().getTotalNumberOfRecords() / new Float(pageSize)));
+            }
+        } else {
+            throw new AuthorityServiceConnectionException(response.getStatusCode().toString());
+        }
+        return searchResult;
+    }
+
+    private String getConceptpowerSearchString(String firstName, String lastName) {
+
+        String conceptpowerSearchString;
+
+        if (firstName != null && lastName != null) {
+
+            conceptpowerSearchString = firstName + "%" + lastName;
```

Comment:
Without "%" it is considering Firstname and Lastname as single string and it was not returning the correct result. After adding this "%" it worked fine as it will consider it separately.

Reply:
Are you sure? Would that create a url like that:
https://chps.asu.edu/conceptpower/rest/ConceptSearch?word=albert%einstein
(which doesn't generate a result)
Do you mean `%20`?

Reply:
Url is generated with "%20"
https://chps.asu.edu/conceptpower/rest/ConceptSearch?word=albert%20einstein

Reply:
In code, I have given "%" sign but the URL is generated with "%20"

Reply:
and what was it generating with " " instead of "%"?

Reply:
It generated a "+" sign between first name and last name and it worked.

Reply:
Not sure why it does that but I don't think that's a requirement of the official RFC (meaning not sure if we can trust that it keeps doing this). Let's make that either a blank (assuming that the url encoder encodes that correctly) or make it directly a `+`.